src/decode.rs

```diff
@@ -4392,19 +4388,11 @@ pub(crate) unsafe fn rav1d_decode_frame_init(
         }
 
         if frame_hdr.allow_screen_content_tools != 0 {
-            if num_sb128 != f.frame_thread.pal_sz {
-                rav1d_freep_aligned(
-                    &mut f.frame_thread.pal as *mut *mut [[u16; 8]; 3] as *mut c_void,
-                );
-                f.frame_thread.pal = rav1d_alloc_aligned(
-                    ::core::mem::size_of::<[[u16; 8]; 3]>() * num_sb128 as usize * 16 * 16,
-                    64,
-                ) as *mut [[u16; 8]; 3];
-                if f.frame_thread.pal.is_null() {
-                    f.frame_thread.pal_sz = 0;
-                    return Err(ENOMEM);
-                }
-                f.frame_thread.pal_sz = num_sb128;
+            if num_sb128 as usize != f.frame_thread.pal_sz() {
+                // TODO: Fallible allocation
+                f.frame_thread
+                    .pal
+                    .resize_with(num_sb128 as usize * 16 * 16, Default::default);
             }
```

Comment:
```suggestion
            // TODO: Fallible allocation
            f.frame_thread
                .pal
                .resize_with(num_sb128 as usize * 16 * 16, Default::default);
```

`.resize_with` will do the check.